osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -79,100 +78,98 @@ protected virtual Track GetVirtualTrack(double emptyLength = 0)
         /// <returns>The applicable <see cref="IBeatmapConverter"/>.</returns>
         protected virtual IBeatmapConverter CreateBeatmapConverter(IBeatmap beatmap, Ruleset ruleset) => ruleset.CreateBeatmapConverter(beatmap);
 
-        public virtual IBeatmap GetPlayableBeatmap(IRulesetInfo ruleset, IReadOnlyList<Mod> mods = null, TimeSpan? timeout = null)
+        public virtual IBeatmap GetPlayableBeatmap(IRulesetInfo ruleset, IReadOnlyList<Mod> mods = null, CancellationToken? cancellationToken = null)
         {
-            using (var cancellationSource = createCancellationTokenSource(timeout))
-            {
-                mods ??= Array.Empty<Mod>();
+            var token = cancellationToken ?? new CancellationTokenSource(10000).Token;
```

Comment:
The timeout should not be applied with a debugger attached, see the original code.

Reply:
I think that should apply to any given cancellation token, not only the timeout default.

Reply:
@frenzibyte let's not scope creep here, shall we? if push comes to shove and a scenario wherein an external token is passed in here needs to be debugged in the future, you could as well replace the token at the root level and debug that.

Reply:
fair enough, have only thought about it from a usability perspective but it's not necessary at the moment anyways.